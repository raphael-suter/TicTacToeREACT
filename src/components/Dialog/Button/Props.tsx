export default interface Props {
  text: string;
  onClick: () => void;
}
