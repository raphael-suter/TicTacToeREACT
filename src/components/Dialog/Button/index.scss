@import "../../../styles/colors.scss";

.button {
  margin: 0.6rem;
  padding: 0.5rem 2rem;
  text-transform: uppercase;
  font-size: 16px;
  border: none;
  background: $mainColor;
  color: $lightColor;
  cursor: pointer;
}
