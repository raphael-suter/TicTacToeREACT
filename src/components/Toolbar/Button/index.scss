@import "../../../styles/colors.scss";

.toolbar-button {
  margin-left: 0.8rem;
  display: flex;
  align-items: center;
  background: $mainColor;
  border: none;
  cursor: pointer;

  span {
    font-size: 24px;
    color: $lightColor;
  }
}
