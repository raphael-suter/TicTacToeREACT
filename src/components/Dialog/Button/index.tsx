import React from 'react';
import './index.scss';
import Props from './Props';

export default ({ text, onClick }: Props) => (
  <button className='dialog-button' onClick={onClick}>{text}</button>
);
