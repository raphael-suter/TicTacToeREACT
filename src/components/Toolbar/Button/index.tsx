import React from 'react';
import './index.scss';
import Props from './Props';

export default ({ icon, onClick }: Props) => (
  <button className='toolbar-button' onClick={onClick}>
    <span className='material-icons'>{icon}</span>
  </button>
);
