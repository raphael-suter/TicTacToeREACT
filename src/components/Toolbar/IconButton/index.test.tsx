import { shallow } from 'enzyme';
import React from 'react';
import Button from '.';

const setup = (icon: string, onClick: () => void) => shallow(<Button icon={icon} onClick={onClick} />);

describe('IconButton', () => {
  test('Renders without crashing.', () => {
    const wrapper = setup(null, null);
    expect(wrapper).toHaveLength(1);
  });

  test('Displays icon.', () => {
    const expectedIcon = 'qwertz';
    const wrapper = setup(expectedIcon, null);
    const receivedIcon = wrapper.text();

    expect(receivedIcon).toBe(expectedIcon);
  });

  test('Fires click event.', () => {
    const onClick = jasmine.createSpy();
    const wrapper = setup(null, onClick);

    wrapper.simulate('click');

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
