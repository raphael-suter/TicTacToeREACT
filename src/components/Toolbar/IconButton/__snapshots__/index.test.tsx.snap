// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton Renders correctly. 1`] = `
<styled.button
  onClick={null}
>
  <styled.span
    className="material-icons"
  >
    delete
  </styled.span>
</styled.button>
`;
